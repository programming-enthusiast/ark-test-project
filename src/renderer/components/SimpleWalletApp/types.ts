export interface Transaction {
	txId: string;
	blockId: string;
	sender: string;
	recipient: string;
	timestamp: string;
	amount: number;
	fee: number;
}

export interface Wallet {
	address: string;
	balance: number;
	transactions: Array<Transaction>;
}

export interface TransactionDataContextType {
	wallets: Array<Wallet>;
	setWallets: (wallets: Array<Wallet>) => void;
	currentWalletIndex: number;
	setCurrentWalletIndex: (index: number) => void;
}
