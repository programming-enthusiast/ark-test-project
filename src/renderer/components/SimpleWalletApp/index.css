@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-Black-slnt=0.ttf") format("truetype");
	font-weight: "black";
}
@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-Bold-slnt=0.ttf") format("truetype");
	font-weight: "bold";
}
@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-ExtraBold-slnt=0.ttf") format("truetype");
	font-weight: "extra-bold";
}
@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-Light-slnt=0.ttf") format("truetype");
	font-weight: "light";
}
@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-Medium-slnt=0.ttf") format("truetype");
	font-weight: "medium";
}
@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-Regular-slnt=0.ttf") format("truetype");
	font-weight: "regular";
}

@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-SemiBold-slnt=0.ttf") format("truetype");
	font-weight: "semi-bold";
}
@font-face {
	font-family: "Inter";
	src: local("Inter"),
		url("./assets/Fonts/Inter-Thin-slnt=0.ttf") format("truetype");
	font-weight: "thin";
}

h1 {
	font-family: "Inter";
	font-weight: "Extra Bold";
	font-size: 36px;
	line-height: 48px;
	color: #1f2121;
}

h2 {
	font-family: "Inter";
	font-weight: "Bold", "Extra Bold";
	font-size: 24px;
	line-height: 36px;
	color: #1f2121;
}

h3 {
	font-family: "Inter";
	font-weight: "Semi Bold", "Bold";
	font-size: 18px;
	line-height: 32px;
	color: #1f2121;
}

.main-font {
	font-family: "Inter";
	font-weight: bold;
	font-size: 16px;
	/* color: #5A5D60; */
}

.auxiliary-font {
	font-family: "Inter";
	font-weight: normal;
	font-size: 14px;
	line-height: 24px;
	color: #5a5d60;
}

.logo-font {
	font-family: "Inter";
	font-weight: bold;
	font-size: 24px;
	color: #ffffff;
}
