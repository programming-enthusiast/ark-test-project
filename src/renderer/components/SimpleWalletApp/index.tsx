import "./index.css";

import React from "react";

import Footer from "./components/Footer";
import Header from "./components/Header";
import TransactionTable from "./components/TransactionTable";
import TransactionDataProvider from "./contexts/TransactionDataProvider";

const SimpleWalletApp = () => (
	<TransactionDataProvider>
		<Header />
		<div className="pt-48px pb-48px">
			<TransactionTable className="w-full" />
		</div>
		<Footer className="" />
	</TransactionDataProvider>
);

export default SimpleWalletApp;
