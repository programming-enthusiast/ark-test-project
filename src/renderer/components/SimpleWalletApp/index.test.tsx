import React from "react";
import renderer from "react-test-renderer";

import SimpleWalletApp from "./";

describe("SimpleWalletApp Component", () => {
	let renderedComponent: any;

	beforeAll(() => {
		renderedComponent = renderer.create(<SimpleWalletApp />);
	});

	it("should render correctly", () => {
		expect(renderedComponent.toJSON()).toMatchSnapshot();
	});
});
