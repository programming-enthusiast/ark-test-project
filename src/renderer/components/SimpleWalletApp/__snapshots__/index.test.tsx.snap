// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleWalletApp Component should render correctly 1`] = `
Array [
  <div
    className="flex h-408px lg:h-212px bg-header-background pt-40px pb-40px px-31px sm:px-31px lg:px-40px"
  >
    <div
      className="flex flex-col lg:flex-row h-312px lg:h-full bg-header-main rounded-3xl w-311px sm:w-577px md:w-577px lg:w-689px xl:w-945px xxl:w-1200px xxxl:w-1200px m-auto"
    >
      <div
        className="my-auto ml-40px lg:m-auto  flex"
      >
        <img
          className="w-56px h-56px lg:m-auto"
          src="logo.svg"
        />
        <h1
          className="logo-font inline m-auto ml-10px block lg:hidden xxl:block"
        >
          ARK Wallet
        </h1>
      </div>
      <div
        className="right-0 top-0 h-2px lg:w-2px lg:h-56px bg-gray-200 mx-40px my-auto lg:m-auto bg-dropdown-background"
      />
      <div
        className="mx-40px my-auto lg:m-auto flex"
        id="wallet-dropdown-menu"
      >
        <div
          className="hidden sm:flex w-51px h-57px rounded-l-full"
          id="wallet-icon"
          style={
            Object {
              "background": "#2F3333",
            }
          }
        >
          <img
            className=" m-auto w-1/3 h-1/3"
            src="address.svg"
          />
          <img
            className="hidden m-auto w-1/3 h-1/3"
            src="address-hover.svg"
          />
        </div>
        <div
          className="custom-select-wrapper w-247px md:w-461px lg:w-557px xl:w-302px xxl:w-404px"
          id="dropdown-menu-wrapper"
        >
          <div
            className="custom-select w-full lg:w-557px xl:w-302px xxl:w-404px "
          >
            <div
              className="custom-select__trigger"
              onClick={[Function]}
            >
              <span
                className="overflow-hidden overflow-ellipsis"
              />
              <div
                className="arrow"
              />
            </div>
            <div
              className="custom-options"
            />
          </div>
        </div>
      </div>
      <div
        className="hidden lg:block right-0 top-0 w-577px h-2px lg:w-2px lg:h-56px bg-gray-200 m-auto bg-dropdown-background"
      />
      <div
        className="mx-40px my-auto lg:m-auto flex"
      >
        <div
          className="flex rounded-full border-dropdown-background border-2 w-57px h-57px"
        >
          <img
            className="w-2/4 h-2/4 m-auto"
            src="balance.svg"
          />
        </div>
        <div
          className="flex flex-col justify-center ml-10px"
        >
          <div
            className="text-font-color2"
          >
            Balance
          </div>
          <div
            className="text-font-color2"
          >
             
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="pt-48px pb-48px"
  >
    <div
      className="text-center"
    >
      No Data
    </div>
  </div>,
  <div
    className="fixed bottom-0 left-0 text-center w-full py-16px bg-footer-background main-font text-font-color1"
  >
    2021 @ ARK.io | All rights reserved
  </div>,
]
`;
